client/components/shipping/packages/input-filters.js

```diff
@@ -0,0 +1,26 @@
+const leftPoint = /^\.\d+$/;
+const rightPoint = /^\d+\.$/;
+const number = ( value ) => {
+	if ( leftPoint.test( value ) ) {
+		value = '0' + value;
+	} else if ( rightPoint.test( value ) ) {
+		value = value.slice( 0, value.length - 1 );
+	}
+
+	return value;
+};
+
+const dimensionRegex = /^(\S+)\s*x\s*(\S+)\s*x\s*(\S+)$/;
+const dimensions = ( value ) => {
+	const result = dimensionRegex.exec( value );
+	if ( result ) {
+		return result.splice( 1, 4 ).map( number ).join( ' x ' );

```

Comment:
I don't know what documentation I was reading, but it said that the first parameter is the start index you want to keep, and the last parameter was the index where you want to stop.

I'm just as baffled as you are about how/why this is working. I'm going to simplify it into something more understandable.